docs/_static/spec/server.yaml

```diff
@@ -0,0 +1,1517 @@
+      description: >-
+        Adds a message to a tracker. This doesn't trigger
+        the prediction loop. It will log the message
+        on the tracker and return, no actions will be
+        predicted or run. This is often used together with the
+        predict endpoint.
+      parameters:
+      - $ref: '#/components/parameters/conversation_id'
+      - $ref: '#/components/parameters/include_events'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                text:
+                  type: string
+                  description: >-
+                    Message text
+                sender:
+                  type: string
+                  description: >-
+                    Origin of the message - who sent it
+                  enum:
+                    - user
+                parse_data:
+                  $ref: '#/components/schemas/ParseData'
+              required: ["message", "sender"]
+            example:
+              sender: user
+              message: hello
+              parse_data:
+                text: hello
+                entities: []
+                intent:
+                  confidence: 0.57
+                  name: greet
+                intent_ranking:
+                - confidence: 0.57
+                  name: greet
+                - confidence: 0.33
+                  name: goodbye
+      responses:
+        200:
+          $ref: '#/components/responses/200Tracker'
+        400:
+          $ref: '#/components/responses/400BadRequest'
+        401:
+          $ref: '#/components/responses/401NotAuthenticated'
+        403:
+          $ref: '#/components/responses/403NotAuthorized'
+        500:
+          $ref: '#/components/responses/500ServerError'
+
+  /models/train:
+    post:
+      security:
+      - TokenAuth: []
+      - JWT: []
+      tags:
+      - Model
+      summary: Train a Rasa Stack model
+      description: >-
+        Trains a Rasa Stack model. A stack model is a model combining a
+        trained dialogue model with an NLU model.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/StackTrainingRequest'
+            example:
+              StackTrainingRequest:
+                $ref: '#/components/examples/StackTrainingRequest'
+      responses:
+        200:
+          description: Zipped Rasa Stack model
+          content:
+            application/octet-stream:
+              schema:
+                $ref: '#/components/schemas/StackTrainingResult'
+        400:
+          $ref: '#/components/responses/400BadRequest'
+        401:
+          $ref: '#/components/responses/401NotAuthenticated'
+        403:
+          $ref: '#/components/responses/403NotAuthorized'
+        500:
+          $ref: '#/components/responses/500ServerError'
+
+  /models/train/nlu:
```

Comment:
do we actually need this separate endpoint or can we combine it with the `train` endpoint? e.g. the commandline scripts are also build to only train the parts we have data for, so if there is no core data we would only train nlu

Reply:
Good point. Would be great if we could combine the training. However, the command line `rasa train` seem to add an "empty" core model to the `tar.gz` file. Loading the model with `rasa run` fails as files are missing in there. Something we need to improve.

Reply:
Mhm, that is odd - can you take a look at it as well?

Reply:
@wochinge I guess that empty model is not intended right?

Reply:
Yes, will add it to the "improve command line interface" issue.

Reply:
> I guess that empty model is not intended right?

No